src/scripts/main.js

```diff
@@ -1,11 +1,30 @@
 'use strict';
 
 function waitFor(element, eventName) {
-  // write your code here
+  return new Promise((resolve) => {
+    element.addEventListener(eventName, () => {
+      resolve(
+        `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`,
+      );
+    });
+  });
 }
 
 const printMessage = (message) => {
-  // write your code here
+  if (document.querySelector('.message') !== null) {
```

Comment:
You are querying the document for an element with the class 'message' twice (lines 14 and 15). It would be more efficient to store the result in a variable and reuse it.